packages/design-system/src/components/icon.tsx

```diff
@@ -1,38 +1,73 @@
 import { type VariantProps, cva } from "class-variance-authority";
-import React from "react";
-import type { IconType } from "../types/icon-types";
-import * as Icons from "./icons";
+import { useId } from "react";
+import { twMerge } from "tailwind-merge";
 
-export const iconVariants = {
+type SvgIconProps = {
```

Comment:
Please reorder the types so that the first type after imports is the "public interface" and it is exported using the same name as the component, plus if you are not reusing the internal definitions (like `SvgIcon`), define it inside the exported type.